test/lib/bamboo/test_adapter_test.exs

```diff
@@ -44,20 +44,107 @@ defmodule Bamboo.TestAdapterTest do
 
     assert_delivered_email sent_email
     refute_delivered_email unsent_email
+
+    sent_email |> TestMailer.deliver_now
+    assert_raise ExUnit.AssertionError, fn ->
+      assert_delivered_email %{sent_email | to: "oops"}
+    end
+  end
+
+  test "assert_delivered_email with no delivered emails" do
+    sent_email = new_email(from: "foo@bar.com", to: ["foo@bar.com"])
+
+    try do
+      assert_delivered_email %{sent_email | to: "oops"}
+    rescue
+      error in [ExUnit.AssertionError] ->
+        assert error.message =~ "0 emails delivered"
+    else
+      _ -> flunk "assert_delivered_email should failed"
```

Comment:
This is necessary so that if the assertion didn't fail we raise. Otherwise it looks like the test passed.